### ISSUE TITLE
clarify documentation about fetchData function

### ISSUE THREAD

**Issue by elray1** [reichlab/predtimechart#44]
- Function arguments are described in the readme under a section titled `App.initialize() args`; it could be helpful to repeat them again below in the `fetchData data format` section
 - In that section, we could add language explaining that for target data, the expected behavior is to return a version of the target data that was available as of the specified `referenceDate`

**Comment by zkamvar** [comment]
It would also be good to reword it to say "target time series data" instead of "truth data".

**Comment by zkamvar** [comment]
Your suggestions would definitely help. 

I think one of the biggest problems that I encountered was that I didn't have any clear examples beyond [the example in predtimechart/src/index.html](https://github.com/reichlab/predtimechart/blob/6678c66fe7a3faab2bcc5186733e7a25cf504d44/src/index.html#L64-L119). Without a narrative behind this example, I could only see that the updated target data was simply adding dates, but nothing else changed:

https://github.com/reichlab/predtimechart/blob/6678c66fe7a3faab2bcc5186733e7a25cf504d44/src/index.html#L107-L117

Since the target data in the flusight hub was a single file, I had assumed that we were just appending new data and not amending existing data (which in hindsight was a foolish assumption, because I even [designed a figure that shows that scenario](https://royalsocietypublishing.org/doi/10.1098/rstb.2018.0276#d1e779)) and just went with it.

Most of the documentation that exists is low-level reference documentation, but for someone without much context to go from, it's not clear where to start or what features are important or auxiliary. A more [explanatory form of documentation](https://diataxis.fr/explanation/) would have been helpful in this situation so that it was clear _where_ the data were coming from and what the context was for the different modes of visualization.
